### PR TITLE
CVE-2018-1305

### PR DESCRIPTION
**Pull request by Bailey-Pearson**
Round 1 Submission. Filling out information for CVE-2018-1305.

### CODE REVIEW

File: cves/CVE-2018-1305.yml [hunk 1]
```diff
@@ -109,9 +114,14 @@ unit_tested:
...
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
-  code:
-  fix:
+  answer: |
+    The code is automatically unit tested before being deployed. This
+    vulnerability was not tested though. They did not change their unit tests during
+    or after the fix. This is probably because they just replaced how they were
+    loading the security constraints and so it didn't need tested as it couldn't
```
Comment (iamnukee) [inline cves/CVE-2018-1305.yml]:
```suggestion
    loading the security constraints and so it didn't need testing as it couldn't
```

File: cves/CVE-2018-1305.yml [hunk 2]
```diff
@@ -109,9 +114,14 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
-  code:
-  fix:
+  answer: |
+    The code is automatically unit tested before being deployed. This
+    vulnerability was not tested though. They did not change their unit tests during
```
Comment (iamnukee) [inline cves/CVE-2018-1305.yml:119]: Do you think the fact that they didn't test this particular part of the code before was one of the reasons why this wasn't found before release?

File: cves/CVE-2018-1305.yml [hunk 3]
```diff
@@ -212,4 +233,12 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    It seems their issue was both a design mistake and a bit of an oversight on 
+    their part. Initally back in 2010 they believed they wanted to be able to
+    deploy servlets at different times. This still is true. But what they didn't
+    think was that in the future the servlets would have secuirty constraints placed
+    on them and this order would be exploitable. They mitigated this risk by forcing
```
Comment (iamnukee) [inline cves/CVE-2018-1305.yml:241]: Do you think this is something they could've foresaw?

File: cves/CVE-2018-1305.yml [hunk 4]
```diff
@@ -74,8 +79,8 @@ fixes:
    - commit:
      note:
 vccs:
-  - commit:
-    note:
+  - commit: ece65c1a428094b1c6c17de3d7593f64e1bb1286
```
Comment (iamnukee) [inline cves/CVE-2018-1305.yml:82]: There were quite a couple fixes. Did these fixes all involve a different step in properly fixing the bug? Or were they attempts and fails until the final one solved the problem?

Comment (Bailey-Pearson) [reply cves/CVE-2018-1305.yml:82]: It seems like the issue was farther reaching than just one file and that is why there was so much changed in the commit

File: cves/CVE-2018-1305.yml (hunk 2 above)
Comment (Bailey-Pearson) [reply cves/CVE-2018-1305.yml:119]: I think without testing you can't know it is broken in the first place but they also probably just didn't think that load order was going to cause a security issue in the first place.

File: cves/CVE-2018-1305.yml (hunk 3 above)
Comment (Bailey-Pearson) [reply cves/CVE-2018-1305.yml:241]: I don't really think so. They would have to have known there would be special security patches for servlets as well as knowing they had to be loaded in a particular order.

**Pull request merged**